src/diagnostics/core_diagnostics.jl

```diff
@@ -606,14 +606,27 @@ compute_pr!(out, state, cache, time) =
 compute_pr!(_, _, _, _, precip_model::T) where {T} =
     error_diagnostic_variable("pr", precip_model)
 
-function compute_pr!(out, state, cache, time, precip_model::Microphysics0Moment)
+function compute_pr!(out, state, cache, time, precip_model::NoPrecipitation)
     if isnothing(out)
-        return cache.precipitation.col_integrated_rain .+
-               cache.precipitation.col_integrated_snow
+        return zeros(Spaces.horizontal_space(axes(state.c.ρ)))
+    else
+        out .= zeros(Spaces.horizontal_space(axes(state.c.ρ)))
+    end
```

Comment:
I would rather define it here, so that I can easily add it to all the simulations in the CI. I.e. my logic was that I want to have a precipitation flux in the plots in the CI for all the simulations. Hopefully soon they will all be running with at least 0M microphysics, but I think now still the default is no precipitation

Reply:
If you want precipitation flux with NoPrecipitation, I think it would be better to define it in the cache and return it for the diagnostics, like what you have for 0M and 1M, rather than having a dispatch on the diagnostics. This way we can avoid the case where someone (accidentally) adds precipitation for NoPrecipitation, but the diagnostics show zero.

Reply:
Sounds good. I'll add that